.github/workflows/ci.yaml

```diff
@@ -69,7 +69,9 @@ jobs:
         run: echo "/usr/lib/jvm/java-17-openjdk-amd64/bin" >> $GITHUB_PATH
 
       - name: Reset existing repo
-        run: git -c "http.https://github.com/.extraheader=" fetch --recurse-submodules=no "https://github.com/lampepfl/dotty" && git reset --hard FETCH_HEAD || true
+        run: |
+          git config --global --add safe.directory /__w/dotty/dotty
+          git -c "http.https://github.com/.extraheader=" fetch --recurse-submodules=no "https://github.com/lampepfl/dotty" && git reset --hard FETCH_HEAD || true
```

Comment:
The `true` at the end made the CI pass when it should have failed (See https://github.com/lampepfl/dotty/actions/runs/7924243033/job/21635455423?pr=19703). Should we remove it ?